include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -43,11 +49,63 @@ namespace __par_backend_hetero
 // aliases for faster access to modes
 using access_mode = sycl::access_mode;
 
+// substitution for C++17 convenience types
 template <typename _T>
 using __decay_t = typename ::std::decay<_T>::type;
 template <bool __flag, typename _T = void>
 using __enable_if_t = typename ::std::enable_if<__flag, _T>::type;
 
+// Bitwise type casting, same as C++20 std::bit_cast
+template <typename _Dst, typename _Src>
+__enable_if_t<
+    sizeof(_Dst) == sizeof(_Src) && ::std::is_trivially_copyable_v<_Dst> && ::std::is_trivially_copyable_v<_Src>, _Dst>
+__dpl_bit_cast(const _Src& __src)
```

Comment:
I just did not think of it. Thanks for noticing.

Reply:
In this particular case, `noexcept` makes no difference because all the code is visible to the compiler. Nevertheless, I added it to match the standard declaration.